### DISCUSSION TITLE
How to make migration to newer versions of templates more straightforward?

### DISCUSSION THREAD

**Discussion by MartinZikmund** [unoplatform/uno#14244] [General]
Currently upgrading an existing solution to new versions of the template can be tricky. Users have three options:

- Compare history on GitHub and apply the changes to the existing solution
- Create a new solution with the same name and compare differences of the default files
- Create a new solution with the same name and copy custom files there

All of these are quite error-prone. Is there a better approach available? Could we have some semi-automated migration tool? The "manual" way would be that each PR in Uno.Templates would also require a change to the release migration docs in Uno repo

**Comment by jeromelaban** [comment]
Templates are not changing that often, once a year in a significant way, but in general, it's not required to upgrade to latest template and the previous structure still works. The only required steps are the ones that are breaking, which are detailed in the migration doc.

Otherwise, if someone wants to update to the latest template structure, create and replace is likely the easiest solution for everyone.

**Reply by MartinZikmund** [reply]
Good example of a fix that would be otherwise missed would be the addition of `Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)"` in the template of the main `.csproj` - small changes like this are easy to miss unless each change is documented 😥 . I personally create a new template and go file-by-file and line-by-line, which is quite time consuming... If each change was at least documented in a single location, it would make it easier to apply them

**Reply by dansiegel** [reply]
While I can appreciate the concern to want to keep things up to date with what we're doing in the template... this is not exactly an easy problem to fix. For one thing how do I know what version of the template you used to create your project... trying update your existing project to what we have out of the box with the current Uno template... without knowing what the starting point was or what customizations you've made / need... this is really an impossible task.
